src/tools/kata-ctl/src/ops/volume_ops.rs

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2022 Boston University
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::args::{DirectVolSubcommand, DirectVolumeCommand};
+
+use anyhow::{anyhow, Ok, Result};
+use futures::executor;
+use hyper;
+use kata_types::mount::DirectVolumeMountInfo;
+use std::{fs, path::Path, path::PathBuf, time::Duration};
+use url;
+
+use runtimes;
+
+const TIMEOUT: Duration = Duration::new(5, 0);
+const CONTENT_TYPE_JSON: &str = "application/json";
+const MOUNT_INFO_FILE_NAME: &str = "mountInfo.json";
+const KATA_DIRECT_VOLUME_ROOT_PATH: &str = "/run/kata-containers/shared/direct-volumes";
+
+pub fn handle_direct_volume(vol_cmd: DirectVolumeCommand) -> Result<()> {
+    let command = vol_cmd.directvol_cmd;
+    match command {
+        DirectVolSubcommand::Add(args) => {
+            add(&args.volume_path, &args.mount_info)?;
+        }
+        DirectVolSubcommand::Remove(args) => {
+            remove(&args.volume_path)?;
+        }
+        DirectVolSubcommand::Stats(args) => {
+            let result = executor::block_on(stats(&args.volume_path))?;
+            println!("{}", &result);
```

Comment:
I guess this is fine for now. But it may be better to make this function return `Result<Option<String>>`. Then, all the commands except this one can return `Result<None>`, but `Stats` can return `Result<Some<String>>`.

You'd need to rename this function and have `handle_direct_volume(...)`:

1. Call the newly named function.
2. check for errors using `?`
3. Unpack the `Result` to get at the `Option` value.
4. If not `None`, print the value.

Structuring the code like that will help with future work to support but logging and various different output formats. It would also mean you can unit test this function more easily.